website/static/js/recentActivityWidget.js

```diff
@@ -0,0 +1,325 @@
+                    } else if ((self.today - self.firstDay)/2629746000 < 3){
+                        self.div = self.div/2;
+                        self.formatFloat = 'MMM Do h';
+                        self.steps = 28;
+                        self.formatPip = 'MMM Do';
+                    } else {
+                        self.formatFloat = 'MMM Do';
+                        self.steps = 31;
+                        self.formatPip = 'MMM';
+                    }
+                }
+                if (reset){
+                    self.totalEvents = result.links.meta.total;
+                    self.eventNumbers = result.links.meta.aggregates;
+                    self.cache = [];
+                }
+                if (!init) {
+                    self.cache.push(result.data.slice(0,9));
+                    self.cache.push(result.data.slice(10,19));
+                    if (!update) {
+                        self.activityLogs(self.cache[self.page - 1]);
+                    }
+                }
+                self.lastPage = (result.links.meta.total / (result.links.meta.per_page/2) | 0) + 1;
+            });
+            return promise;
+        };
+
+        self.callLogs = function(filter) {
+            self.eventFilter = self.eventFilter === filter ? false : filter;
+            self.page = 1;
+            self.cache = [];
+            self.getLogs();
+        };
+
+        self.getLogs(true, false);
+
+        $(window).resize(function(){
+            var canvas = document.getElementById('rACanvas');
+            var ctx = canvas.getContext('2d');
+            ctx.clearRect(0, 0, canvas.width, canvas.height);
+        });
+
+        self.makeLine = function(canvas){
+            var ctx = canvas.getContext('2d');
+            ctx.clearRect(0, 0, canvas.width, canvas.height);
+            var progBar = $('#rAProgressBar');
+            var handle = $('.ui-slider-handle');
+            var leftHandle = handle[0];
+            var rightHandle = handle[1];
+            ctx.beginPath();
+            ctx.moveTo(leftHandle.offsetLeft + (handle.width()/2), 0);
+            ctx.lineTo(progBar.offset().left - $('#rACanvas').offset().left, self.canvasHeight);
+            ctx.strokeStyle = '#E0E0E0 ';
+            ctx.lineWidth = 2;
+            ctx.stroke();
+            ctx.beginPath();
+            ctx.moveTo(rightHandle.offsetLeft + (handle.width()/2), 0);
+            ctx.lineTo(progBar.offset().left + progBar[0].offsetWidth - $('#rACanvas').offset().left, self.canvasHeight);
+            ctx.strokeStyle = '#E0E0E0 ';
+            ctx.lineWidth = 2;
+            ctx.stroke();
+        };
+        self.categoryColor = function(category){
+            if (category.indexOf('wiki') !== -1){ return '#d9534f'; }
+            if (category.indexOf('comment') !== -1){ return '#5bc0de'; }
+            if (category.indexOf('file') !== -1){ return '#337ab7'; }
+            if (category.indexOf('project') !== -1){ return '#f0ad4e'; }
```

Comment:
Yes that's correct; as long as this method catches all relevant instances this method works fine. For instance every file category has an action that has the word file in them.